depends/builders/default.mk

```diff
@@ -8,9 +8,9 @@ default_build_OTOOL = otool
 default_build_INSTALL_NAME_TOOL = install_name_tool
 
 define add_build_tool_func
-ifeq ($(filter $(origin $1),undefined default),)
-build_$(build_os)_$1 = $(or $($1),$(build_$(build_os)_$1),$(default_build_$1))
-build_$(build_arch)_$(build_os)_$1 = $(or $($1),$(build_$(build_arch)_$(build_os)_$1),$$(build_$(build_os)_$1))
+ifneq ($(origin $1_FOR_BUILD),undefined)
+build_$(build_os)_$1 = $(or $($1_FOR_BUILD),$(build_$(build_os)_$1),$(default_build_$1))
+build_$(build_arch)_$(build_os)_$1 = $(or $($1_FOR_BUILD),$(build_$(build_arch)_$(build_os)_$1),$$(build_$(build_os)_$1))
```

Comment:
Perhaps we should establish a style for Makefiles... But I find it more explicit to say `$($(1)_FOR_BUILD)` like we do for `$($(1)_postprocessed)` etc.

Reply:
Sorry Carl, but I'm not sure if I understand you correctly...

If you mean s/`$($1_FOR_BUILD)`/`$($(1)_FOR_BUILD)`/, both are not equivalent though.

Reply:
Oh, then in that case shouldn't we be using `$($(1)_FOR_BUILD)` in these cases for correctness' sake?

Reply:
> Oh, then in that case shouldn't we be using `$($(1)_FOR_BUILD)` in these cases for correctness' sake?

We are interested in `$(CXX_FOR_BUILD)` rather `$(g++_FOR_BUILD)`, no?

Reply:
I was definitely wrong. Sorry for noise.

[Updated](https://github.com/bitcoin/bitcoin/pull/23571#issuecomment-998360080).